test/Libraries/DynamoMSOfficeTests/ExcelTests.cs

```diff
@@ -983,6 +979,247 @@ public void TestWriteFunctionObjectToExcel()
+
+            // remap the filename as Excel requires an absolute path
+            filename.Value = filename.Value.Replace(@"..\..\..\test", TestDirectory);
+
+            var watch = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace("8651eeba-33a9-4704-b92c-346e6e5e4b34");
+            ViewModel.HomeSpace.Run();
+            Assert.IsTrue(watch.CachedValue.IsCollection);
+            var data = new object[] { new object[] { 1 }, new object[] { 2 }, new object[] { 3 }, new object[] {4} ,new object []{5}};
+            AssertPreviewValue(watch.GUID.ToString(), data);
+           
+        }
+        [Test]
+        public void WriteNonExistingSheet()
+        {
+
+            // 1.  Open Sample ImportExport_Excel_ToDynamo.dyn and Run
+            // 2.  NonExisting Sheet
+            //3.  Rerun the file  
+            string openPath = Path.Combine(TestDirectory, @"core\excel\WriteNonExistingSheet.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
+
+            var filename = ViewModel.Model.CurrentWorkspace.FirstNodeFromWorkspace<DSCore.File.Filename>();
+
+            // remap the filename as Excel requires an absolute path
+            filename.Value = filename.Value.Replace(@"..\..\..\test", TestDirectory);
+
+            var watch = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace("32dbe300-4780-4c47-b07c-63019d5285ac");
+            ViewModel.HomeSpace.Run();
+            Assert.IsTrue(watch.CachedValue.IsCollection);
+           
+            var data = new object[] { new object[] { 1 }, new object[] { 2 }, new object[] { 3 }, new object[] { 4 }, new object[] { 5 } };
+            AssertPreviewValue(watch.GUID.ToString(), data);
+        }
+       
+        public void WritenodeWithWarning()
+        {
+            // Write a node with warning to Excel.
+            // If there is any data in excel it must not overwrite here- currently this fails
+            string openPath = Path.Combine(TestDirectory, @"core\excel\WriteNodewithError.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
+
+            var filename = ViewModel.Model.CurrentWorkspace.FirstNodeFromWorkspace<DSCore.File.Filename>();
+
+            // remap the filename as Excel requires an absolute path
+            filename.Value = filename.Value.Replace(@"..\..\..\test", TestDirectory);
+            var watch = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace("32dbe300-4780-4c47-b07c-63019d5285ac");
+            ViewModel.HomeSpace.Run();
+            
+            var list = watch.CachedValue.GetElements();
+
+            // input array is empty list, {}
+            // Ensure output is also empty list
+            Assert.AreEqual(0, list.Count);
+
+            Assert.IsTrue(watch.CachedValue.IsCollection);
+            Assert.IsNull(watch.CachedValue.Data);
+   
+           
+        }
+       [Test]
+        public void WriteToReadOnlyFile()
```

Comment:
Unfortunately these read only excel files you are submitting are not guaranteed to be read only and so can fail on different machines.